lib/resty/upstream/healthcheck.lua

```diff
@@ -92,6 +93,71 @@ local function set_peer_down_globally(ctx, is_backup, id, value)
     end
 end
 
+local function get_all_peers(u)
+    local primary_peers, err = upstream.get_primary_peers(u)
+    if not primary_peers then
+        errlog("failed to get primary peers: ", err)
+        return
+    end
+
+    local backup_peers, err = upstream.get_backup_peers(u)
+    if not backup_peers then
+        errlog("failed to get backup peers: ", err)
+        return
+    end
+
+    return primary_peers, backup_peers
+end
+
+local function select_and_check_unchecked_peers(peers, u)
+  local filtered_peers = {}
+  for _, peer in ipairs(peers) do
+    if not checked_peers_upstream[peer.name] then
+      checked_peers_upstream[peer.name] = u
+      table.insert(filtered_peers, peer)
+    end
+  end
+  return filtered_peers
+end
+
+local function copy_status_if_checked(peers, is_backup, ctx)
+    local checked_peers = {}
+    for _, peer in ipairs(peers) do
+        if checked_peers_upstream[peer.name] then
+            table.insert(checked_peers, peer)
+        end
+    end
+
+    local dict = ctx.dict
+    local u = ctx.upstream
+
+    for _, peer in ipairs(peers) do
+        local id = peer.id
+        local already_set = dict:get(gen_peer_key("ok:", u, is_backup, id))
+        if already_set then
+            return
+        end
+
+        local checked_peer_upstream = checked_peers_upstream[peer.name]
+
+        local oks, _ = dict:get(gen_peer_key("ok:", checked_peer_upstream, is_backup, id))
+        if oks then
+            dict:set(gen_peer_key("ok:", u, is_backup, id), 1)
+        end
+
+        local noks, _ = dict:get(gen_peer_key("nok:", checked_peer_upstream, is_backup, id))
+        if noks then
+            dict:set(gen_peer_key("nok:", u, is_backup, id), 1)
+        end
+    end
+end
+
+local function copy_status(ctx)
+    local ppeers_checked, bpeers_checked = get_all_peers(ctx.upstream)
```

Comment:
These aren't necessarily checked yet, so it's more correct to call them `ppeers, bpeers`.